### PR TITLE
unified repo (app-spec)

### PR DESCRIPTION
**Pull request by zippy**
moves app-spec into holochain-rust

- adds `make test_app_spec` to Makefile
- adds app_spec section to travis
- updates readmes

### CODE REVIEW

File: app-spec/docker/Dockerfile.app-spec-rust [hunk 1]
```diff
@@ -0,0 +1,46 @@
+FROM holochain/holochain-rust:develop
```
Comment (Connoropolous) [inline app-spec/docker/Dockerfile.app-spec-rust]: Prbly don't need this file in here now

File: cmd/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "hc"
-version = "0.0.2"
+version = "0.0.3"
```
Comment (Connoropolous) [inline cmd/Cargo.toml]: why is this version number getting bumped here?

Comment (zippy) [reply cmd/Cargo.toml]: I just did that at one point to test that it was getting compliled will fix.

File: app-spec/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+This repository contains a Holochain app that functions as a living specification of Holochain and its Rust [HDK](https://github.com/holochain/holochain-rust/tree/develop/hdk-rust) (Holochain Development Toolkit).
+
+As new features, or changes to the HDK (and the API) are being designed, they will be made concrete by adding a use case to this example app and putting those changes as a pull-request to this repository. As soon as the current master branch in https://github.com/holochain/holochain-rust implements all used features, the PR gets merged here so that this repository's master branch always reflects the feature set available in Holochain's master branch.
+
+Please see the [Contribute section](https://github.com/holochain/holochain-rust/blob/develop/README.md#app-spec-driven-development) for our protocol on how we do this.
+
+### Dependencies
+
+The primary dependency is on the Holochain command line tools. These can be installed here:
+* https://github.com/holochain/holochain-cmd
```
[the remark this answers is not in the document]
Comment (zippy) [reply app-spec/README.md]: yep, have to fix readmes...

File: app-spec/test/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "devDependencies": {
+    "@babel/core": "^7.0.0",
```
Comment (Connoropolous) [inline app-spec/test/package.json]: ALL the devDependencies except faucet can/should come out

Comment (zippy) [reply app-spec/test/package.json]: done

**Comment by Connoropolous** [comment]
fantasstic!

**Pull request merged**